lib/posts.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+
+import 'users.dart';
+import 'models/post_model.dart';
+
+class FahadPosts {
+  static const PostModel persona = PostModel(
+    "The new Persona 3 remake is AWESOME",
+    user: Users.fahad,
+    likes: 6,
+    favorites: 2,
+    dislikes: 1,
+    imageAttachment: AssetImage("assets/posts/persona.jpg"),
+  );
+  static const PostModel flutter = PostModel(
+    "Learning flutter's been pretty cool",
+    user: Users.fahad,
+    likes: 2,
+  );
+}
```

Comment:
Good idea to separate the posts, but I recommend you create a field in `UserModel` with name `posts` and attach each user posts with it, with that you can access the posts easily for time being.

Read about encapsulation
https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)

Reply:
So just a little commentary on this.

I remember now that I had initially thought of this when I was writing the code, but I thought that it would be an unnecessary performance (specifically memory) overhead to associate each `UserModel` with a list of posts that will probably not be used for the use context of `UserModel` (assuming we do this by having a `List<PostModel>` field in the `UserModel` class). Things like post reactions, as well as individual replies under certain posts don't need to load *every* post of that user in memory, when it's overwhelmingly likely that most of it is not needed nor will it be used.

In retrospect, this issue would only really be an issue if it isn't the case that the `posts` variable wouldn't necessarily be the raw contents of the posts, but something that points to the list of posts. Such that, once we load any given `UserModel` variable, we're not loading all the posts, but we're loading the pointer or access point, if you will, of the list of posts. I don't know how Dart works under the hood, but it's probably likely that this is how it's done since almost all languages allocate the contents of dynamic arrays on the heap, and not in the same area where the List variable actually exists.

So your approach is probably right, but I just seemed to have overthought the performance implications of having a List variable inside a class in Dart, and overlooked how dynamic arrays work for most languages :)

If you're interested more in this topic, here are some great resources for learning about it:
- Mike Acton's (*brilliant*) [talk](https://youtu.be/rX0ItVEVjHc?si=bR-S2gDUydWCaQFe) about the performance implications of poorly structured data types.
- Core Dumped's great videos explaining [how the heap memory segment works](https://youtu.be/ioJkA7Mw2-U?si=JyZEXV9kxSJ52Te2) and [how most languages (including JavaScript) handle dynamic arrays in memory](https://youtu.be/xFMXIgvlgcY?si=5i2f8d5Pwgc4AwMn)

And of course, thank you Hassan and Arwa for your leadership of the GDSC dev unit :)